src/renderer/components/ReactFlowBox.tsx

```diff
@@ -42,10 +43,59 @@ import { SettingsContext } from '../contexts/SettingsContext';
 import { DataTransferProcessorOptions, dataTransferProcessors } from '../helpers/dataTransfer';
 import { AABB, Point, getBezierPathValues, pointDist } from '../helpers/graphUtils';
 import { isSnappedToGrid, snapToGrid } from '../helpers/reactFlowUtil';
+import { useHotkeys } from '../hooks/useHotkeys';
+import { useIpcRendererListener } from '../hooks/useIpcRendererListener';
 import { useMemoArray } from '../hooks/useMemo';
 import { useNodesMenu } from '../hooks/useNodesMenu';
 import { usePaneNodeSearchMenu } from '../hooks/usePaneNodeSearchMenu';
 
+const elk = new ELK();
+
+const elkOptions = {
+    'elk.algorithm': 'layered',
+    'elk.layered.spacing.nodeNodeBetweenLayers': '100',
+    'elk.spacing.nodeNode': '80',
+    'elk.direction': 'RIGHT',
+};
+
+// This is basically just copypasta'd from the RF docs
+// I just modified it a bit
+const getLayoutedElements = async (nodes: Node<NodeData>[], edges: Edge<EdgeData>[]) => {
+    const isHorizontal = elkOptions['elk.direction'] === 'RIGHT';
+    const graph: ElkNode = {
+        id: 'root',
+        layoutOptions: elkOptions,
+        children: nodes.map((node) => ({
+            ...node,
+            // Adjust the target and source handle positions based on the layout
+            // direction.
+            targetPosition: isHorizontal ? 'left' : 'top',
+            sourcePosition: isHorizontal ? 'right' : 'bottom',
+
+            // Hardcode a width and height for elk to use when layouting.
+            width: node.width || 150,
+            height: node.height || 50,
+        })),
+        edges: edges.map((edge) => ({
+            ...edge,
+            sources: [edge.source],
+            targets: [edge.target],
+        })),
+    };
+
+    const layoutedGraph = await elk.layout(graph);
+    return {
+        nodes: layoutedGraph.children?.map((node) => ({
+            ...node,
+            // React Flow expects a position property on the node instead of `x`
+            // and `y` fields.
+            position: { x: node.x, y: node.y },
+        })),
+
+        edges: layoutedGraph.edges,
```

Comment:
The layout of our edges is determined solely by the connected nodes. So why do you return this here?

Reply:
Copypasta boilerplate